lib/improver/nowcasting/accumulation.py

```diff
@@ -133,23 +180,119 @@ def _get_period_sets(self, time_interval, cubes):
                    "total accumulation period.".format(
                        self.accumulation_period, time_interval))
             raise ValueError(msg)
+        return cubes, time_interval, integral
+
+    def _get_cube_subsets(self, cubes, cube, integral):
+        """Finding the subset of cubes from the input cubelist that are
+        within the accumulation period, based on the input cube defining the
+        upper bound of the accumulation period and the length of the
+        accumulation period.
+
+        Args:
+            cubes (iris.cube.CubeList):
+                Cubelist containing all the rates cubes that will be used
+                to calculate the accumulation.
+            cube (iris.cube.Cube):
+                Cube defining the upper bound of the accumulation period.
+            integral (int)
+                Integer value based on how many times the accumulation
+                period is divisible by the time interval. This therefore
+                indicates how many timesteps are expected within an
+                accumulation period.
+
+        Returns:
+            cube_subset (iris.cube.Cubelist) or None:
+                Cubelist that defines the cubes used to calculate
+                the accumulations.
 
-        cube_subset = int(cube_subset)
+        Warns:
+            Warning: The provided cubes result in a partial period given the
+                specified accumulation_period, i.e. the number of cubes is
+                insufficient to give a set of complete periods. Only complete
+                periods will be returned.
+        """
+        fp_point = cube.coord("forecast_period").points[0]
+        if fp_point not in self.forecast_periods:
+            return None
```

Comment:
This seems rather back-to-front. Why not loop over the thing that is wanted (forecast period as a np.arange object in the process method) rather than the thing that you have (all the cubes in the cube-list, rejecting any that don't happen to match).

Reply:
As suggested, I've now swapped this, so the iteration is over forecast period, rather than cube, so as to avoid any redundant iteration.